soaks/common/terraform/modules/lading_splunk_hec_blackhole/main.tf

```diff
@@ -0,0 +1,112 @@
+resource "kubernetes_config_map" "lading" {
+  metadata {
+    name      = "lading-splunk-hec-blackhole"
+    namespace = var.namespace
+  }
+
+  data = {
+    "splunk_hec_blackhole.toml" = var.splunk-hec-blackhole-toml
+  }
+}
+
+resource "kubernetes_service" "splunk-hec-blackhole" {
+  metadata {
+    name      = "splunk-hec-blackhole"
+    namespace = var.namespace
+  }
+  spec {
+    selector = {
+      app  = "splunk-hec-blackhole"
+      type = var.type
+    }
+    session_affinity = "ClientIP"
+    port {
+      name        = "prom-export"
+      port        = 9090
+      target_port = 9090
+    }
+    type = "ClusterIP"
+  }
+}
+
+resource "kubernetes_deployment" "splunk-hec-blackhole" {
+  metadata {
+    name      = "splunk-hec-blackhole"
+    namespace = var.namespace
+    labels = {
+      app  = "splunk-hec-blackhole"
+      type = var.type
+    }
+  }
+
+  spec {
+    replicas = 1
+
+    selector {
+      match_labels = {
+        app  = "splunk-hec-blackhole"
+        type = var.type
+      }
+    }
+
+    template {
+      metadata {
+        labels = {
+          app  = "splunk-hec-blackhole"
+          type = var.type
+        }
+        annotations = {
+          "prometheus.io/scrape" = true
+          "prometheus.io/port"   = 9090
+          "prometheus.io/path"   = "/metrics"
+        }
+      }
+
+      spec {
+        automount_service_account_token = false
+        container {
+          image_pull_policy = "IfNotPresent"
+          image             = "ghcr.io/blt/lading:0.5.3"
```

Comment:
I was thinking if we wanted to be able to adjust this more easily than having to update each module, curious if anyone has thoughts.

Reply:
Maybe a shared variables file that has a default?

Reply:
Yeah, I've been bitten in the past where a bunch of modules rely on a single item which an update to breaks a number of the depending modules - but it's also nice to be able to push updates in a single spot.

Reply:
@blt do you have an opinion here? I'm a little rusty with my terraform but it _might_ need to be like another module that gets pulled in by everything else 💤 `/common/terraform/common/` 💤

Reply:
Oh nice yeah - I wasn't onboard until the "pass through the common shell script entrypoint" because per test would have been the same as per module. I think that's a good plan 👍

Reply:
Ah, I see, thanks! Will give that a shot.

Reply:
If you like! Definitely not blocking.

Reply:
Gotcha, I'll follow up on this work in another PR to better separate concerns.